netket/utils/group/_permutation_group.py

```diff
@@ -23,35 +23,79 @@
 from netket.utils.types import Array, DType, Shape
 from netket.utils.dispatch import dispatch
 
+from netket.utils import warn_deprecation, deprecated
+
 from ._group import FiniteGroup
 from ._semigroup import Element
 
 
 class Permutation(Element):
-    def __init__(self, permutation: Array, name: str | None = None):
+    def __init__(
+        self,
+        permutation: Array | None = None,
+        *,  # change one line somewhere
+        name: str | None = None,
+        permutation_array: Array | None = None,
+        inverse_permutation_array: Array | None = None,
+    ):
         r"""
-        Creates a `Permutation` from an array of preimages of :code:`range(N)`.
+        Creates a `Permutation` from either the array of images `permutation_array` or preimages `inverse_permutation_array`. Note that the left action of a permutation on an array `a` is `a[inverse_permutation_array]`.
 
-        Arguments:
-            permutation: 1D array listing :math:`g^{-1}(x)` for all :math:`0\le x < N`
-                (i.e., :code:`V[permutation]` permutes the elements of `V` as desired)
-            name: optional, custom name for the permutation
+        Args:
+            permutation: (deprecated) 1D array listing :math:`g^{-1}(x)` for all :math:`0\le x \le N-1`.
+            name: Optional, custom name for the permutation.
+            permutation_array: 1D array listing :math:`g(x)` for all :math:`0\le x \le N-1`.
+            inverse_permutation_array: 1D array listing :math:`g^{-1}(x)` for all :math:`0\le x \le N-1`.
 
         Returns:
-            a `Permutation` object encoding the same permutation
+            A `Permutation` object encoding the same permutation.
         """
-        self.permutation = HashableArray(np.asarray(permutation))
+
+        arg_list = [permutation, permutation_array, inverse_permutation_array]
+        if sum([arg is not None for arg in arg_list]) != 1:
+            raise TypeError(
+                "Only one among `permutation`, `permutation_array` and `inverse_permutation_array` must be specified."
```

Comment:
```suggestion
                "Precisely one of `permutation`, `permutation_array`,\n"
                "and `inverse_permutation_array` must be specified."
```
not specifying any of them is also bad